### PR TITLE
add nigeria oyo state spider

### LINKED ISSUE #851: Nigeria Oyo State spider
**Links to bulk download and/or API endpoints**
https://ocdsoyo.oyostate.gov.ng/json_formation.php from https://ocdsoyo.oyostate.gov.ng/projects.php

**Link to CRM issue**
CRM-7735

**Priority**
anytime

**Data structure**
A release-package, DRT link of the latest published data shows no structural errors

*If the publication passes the [basic criteria](https://standard.open-contracting.org/latest/en/getting_started/quality/#basic-criteria), please inform Yohanna and/or James to add it to the data registry.*
The publication passes the basic criteria

### PR DESCRIPTION
**Pull request by Ravf95**
Closes #851

### CODE REVIEW

File: kingfisher_scrapy/spiders/nigeria_oyo_state.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import scrapy
+
+from kingfisher_scrapy.base_spider import SimpleSpider
+
+
+class NigeriaOyoStateSpider(SimpleSpider):
```
Comment (yolile) [inline kingfisher_scrapy/spiders/nigeria_oyo_state.py:6]:
```suggestion
class NigeriaOyoState(SimpleSpider):
```

File: docs/spiders.rst [hunk 2]
```diff
@@ -1011,6 +1011,13 @@ Nigeria
 
    scrapy crawl nigeria_kaduna_state_releases
 
+.. autoclass:: kingfisher_scrapy.spiders.nigeria_oyo_state.NigeriaOyoStateSpider
```
Comment (yolile) [inline docs/spiders.rst:1014]:
```suggestion
.. autoclass:: kingfisher_scrapy.spiders.nigeria_oyo_state.NigeriaOyoState
```

**Review by yolile — CHANGES_REQUESTED** [review]

File: kingfisher_scrapy/spiders/nigeria_oyo_state.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    """
+    name = 'nigeria_oyo_state'
+
+    # SimpleSpider
+    data_type = 'release_package'
+
+    def start_requests(self):
+        # From https://ocdsoyo.oyostate.gov.ng/projects.php
+        url = 'https://ocdsoyo.oyostate.gov.ng/json_formation.php'
+        yield scrapy.Request(url, meta={'file_name': 'all.json'})
```
Comment (yolile) [inline kingfisher_scrapy/spiders/nigeria_oyo_state.py:19]:
```suggestion
        yield scrapy.Request('https://ocdsoyo.oyostate.gov.ng/json_formation.php', meta={'file_name': 'all.json'})
```

File: kingfisher_scrapy/spiders/nigeria_oyo_state.py (hunk 1 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/nigeria_oyo_state.py:6]: Done

File: docs/spiders.rst (hunk 2 above)
Comment (Ravf95) [reply docs/spiders.rst:1014]: Done

File: kingfisher_scrapy/spiders/nigeria_oyo_state.py (hunk 3 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/nigeria_oyo_state.py:19]: Done

**Review by yolile — APPROVED** [review]

**Pull request merged**